pkg/state/state.go

```diff
@@ -24,30 +30,145 @@ func NewStateClient(fs filesystem.Filesystem) *State {
 	}
 }
 
-func (o *State) SetForwardedPorts(fwPorts []api.ForwardedPort) error {
+func (o *State) Init(ctx context.Context) error {
+	var (
+		pid      = odocontext.GetPID(ctx)
+		platform = fcontext.GetPlatform(ctx, commonflags.PlatformCluster)
+	)
+	o.content.PID = pid
+	o.content.Platform = platform
+	return o.save(ctx, pid)
+
+}
+
+func (o *State) SetForwardedPorts(ctx context.Context, fwPorts []api.ForwardedPort) error {
+	var (
+		pid      = odocontext.GetPID(ctx)
+		platform = fcontext.GetPlatform(ctx, commonflags.PlatformCluster)
+	)
 	// TODO(feloy) When other data is persisted into the state file, it will be needed to read the file first
 	o.content.ForwardedPorts = fwPorts
-	return o.save()
+	o.content.PID = pid
+	o.content.Platform = platform
+	return o.save(ctx, pid)
 }
 
-func (o *State) GetForwardedPorts() ([]api.ForwardedPort, error) {
-	err := o.read()
-	if err != nil {
-		if errors.Is(err, fs.ErrNotExist) {
-			return nil, nil // if the state file does not exist, no ports are forwarded
+func (o *State) GetForwardedPorts(ctx context.Context) ([]api.ForwardedPort, error) {
+	var (
+		result    []api.ForwardedPort
+		platforms []string
+		platform  = fcontext.GetPlatform(ctx, "")
+	)
+	if platform == "" {
+		platforms = []string{commonflags.PlatformCluster, commonflags.PlatformPodman}
+	} else {
+		platforms = []string{platform}
+	}
+
+	for _, platform = range platforms {
+		content, err := o.read(platform)
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				continue // if the state file does not exist, no ports are forwarded
+			}
+			return nil, err
 		}
-		return nil, err
+		result = append(result, content.ForwardedPorts...)
 	}
-	return o.content.ForwardedPorts, err
+	return result, nil
 }
 
-func (o *State) SaveExit() error {
+func (o *State) SaveExit(ctx context.Context) error {
+	var (
+		pid = odocontext.GetPID(ctx)
+	)
 	o.content.ForwardedPorts = nil
-	return o.save()
+	o.content.PID = 0
+	o.content.Platform = ""
+	err := o.delete(pid)
+	if err != nil {
+		return err
+	}
+	return o.saveCommonIfOwner(pid)
 }
 
 // save writes the content structure in json format in file
-func (o *State) save() error {
+func (o *State) save(ctx context.Context, pid int) error {
+
+	err := o.checkFirstInPlatform(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = o.saveCommonIfOwner(pid)
+	if err != nil {
+		return err
+	}
+
+	jsonContent, err := json.MarshalIndent(o.content, "", " ")
+	if err != nil {
+		return err
+	}
+	// .odo directory is supposed to exist, don't create it
+	dir := filepath.Dir(getFilename(pid))
+	err = os.MkdirAll(dir, 0750)
+	if err != nil {
+		return err
+	}
+	return o.fs.WriteFile(getFilename(pid), jsonContent, 0644)
```

Comment:
Could we extract a separate method from L108 through L118? I saw this repeated, and the only difference was the target state file path we are writing to.